Description: Fixes scripts/set-hostname for Debian
 scripts/set-hostname was designed for CentOS, modifying /etc/sysconfig/network.
 This patch fixes that and makes it work everywhere.
Author: Thomas Goirand <zigo@debian.org>
Forwarded: no

--- xen-api-1.3.2.orig/scripts/set-hostname
+++ xen-api-1.3.2/scripts/set-hostname
@@ -10,9 +10,13 @@ fi
 HOSTNAME=$1
 
 # Update system configuration
-sed -i -e "s/^\(HOSTNAME=\).*$/\1$1/g" /etc/sysconfig/network
-
-# Set current hostname
-hostname "$HOSTNAME"
+if [ -f /etc/debian_version ] ; then
+	echo $1 >/etc/hostname
+	invoke-rc.d hostname.sh start
+else
+	sed -i -e "s/^\(HOSTNAME=\).*$/\1$1/g" /etc/sysconfig/network
+	# Set current hostname
+	hostname "$HOSTNAME"
+fi
 
 exit 0
